src/runtime_src/core/edge/skd/xrt_skd.cpp

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright (C) 2022, Xilinx Inc - All rights reserved
+ * Xilinx Runtime (XRT) Experimental APIs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "xrt_skd.h"
+
+namespace xrt {
+
+  /**
+   * skd() - Constructor from uuid and soft kernel section
+   * 
+   * @param kernel metadata buffer handle
+   *
+   * @param soft kernel image buffer handle
+   *
+   * @param soft kernel name
+   *
+   * @param soft kernel CU index
+   *
+   */
+  skd::skd(xclDeviceHandle handle, int sk_meta_bohdl, int sk_bohdl, char *kname, uint32_t cu_index, char *uuid) {
+    strcpy(sk_name,kname);
+    devHdl = handle;
+    xrtdHdl = xrt::device(handle);
+    cu_idx = cu_index;
+    sk_bo = sk_bohdl;
+    sk_meta_bo = sk_meta_bohdl;
+    memcpy(xclbin_uuid,uuid,sizeof(xclbin_uuid));
+
+    // Set sk_path according to sk_name
+    snprintf(sk_path, XRT_MAX_PATH_LENGTH, "%s%s%d", SOFT_KERNEL_FILE_PATH,
+	     sk_name,cu_idx);
+  }
+
+  XCL_DRIVER_DLLESPEC
+  int
```

Comment:
Being that this is C++ code, we should be using exceptions instead of return values to indicate failures.